activitysim/abm/models/settings_checker.py

```diff
@@ -0,0 +1,158 @@
+) -> DataFrame:
+    logger.info(
+        f"Attempting to load SPEC for {model_name} via {model_settings.__class__.__name__}"
+    )
+    spec_file = model_settings.model_dump().get("SPEC")
+    if spec_file is None:
+        logger.info(
+            f"No SPEC file is associated with {model_settings.__class__.__name__}"
+        )
+    spec = state.filesystem.read_model_spec(spec_file)
+    logger.info(f"Successfully loaded model SPEC from {spec_file}")
+    return spec
+
+
+def try_load_and_eval_coefs(
+    model_name: str, model_settings: PydanticBase, spec: DataFrame, state: State
+) -> tuple[DataFrame | None, DataFrame | None]:
+    logger.info(
+        f"Attempting to load coefficients for {model_name} via {model_settings.__class__.__name__}"
+    )
+    if hasattr(model_settings, "COEFFICIENTS"):
+        coefs_file = model_settings.COEFFICIENTS
+        coefs = state.filesystem.read_model_coefficients(model_settings)
+        # check whether coefficients should be evaluated as NESTS or not
+        if model_settings.model_dump().get("NESTS"):
+            # TODO: determine appropriate value of trace_label?
+            # FIXME: is this the correct way to invoke this?
+            eval_coefs = eval_nest_coefficients(
+                model_settings.NESTS, coefs, trace_label=None
+            )
+        else:
+            eval_coefs = eval_coefficients(state, spec, coefs, estimator=None)
+        logger.info(f"Successfully read and evaluated coefficients from {coefs_file}")
+        return coefs, eval_coefs
+    else:
+        logger.info(
+            f"No coefficients file is associated with {model_settings.__class__.__name__}"
+        )
+    return None, None
+
+
+def try_load_and_check(
+    model_name: str,
+    model_settings_class: Type[PydanticBase],
+    model_settings_file: str,
+    state: State,
+) -> None:
+
+    # first, attempt to load settings
+    settings = try_load_model_settings(
+        model_name=model_name,
+        model_settings_class=model_settings_class,
+        model_settings_file=model_settings_file,
+        state=state,
+    )
+
+    # then, attempt to read SPEC file
+    spec = try_load_spec(model_name=model_name, model_settings=settings, state=state)
+
+    # then, attempt to read and evaluate coefficients
+    coefs, eval_coefs = try_load_and_eval_coefs(
+        model_name=model_name, model_settings=settings, spec=spec, state=state
+    )
+
+    # then, check preprocessors if any
+    # for now, check is limited to check that the SPEC file is loadable
+    if settings.model_dump().get("preprocessor"):
+        preprocessor_settings = settings.preprocessor
+        spec = try_load_spec(
+            model_name=model_name + ": preprocessor",
+            model_settings=preprocessor_settings,
+            state=state,
+        )
+
+
+def check_model_settings(state: State) -> None:
+
+    components = state.settings.models  # _RUNNABLE_STEPS.keys() may be better?
```

Comment:
state.settings.models is probably the right thing here not `_RUNNABLE_STEPS`, as the latter includes every step that might be run, but the former includes only that which we want to run right now.  It's totally legit to have incomplete/invalid settings for steps we are not trying to run right now.